jetty-io/src/main/java/org/eclipse/jetty/io/ManagedSelector.java

```diff
@@ -244,20 +246,31 @@ private int getActionSize()
     @Override
     public void dump(Appendable out, String indent) throws IOException
     {
-        super.dump(out, indent);
         Selector selector = _selector;
+        List<String> keys = null;
+        List<Runnable> actions = null;
         if (selector != null && selector.isOpen())
         {
-            List<Runnable> actions;
+            DumpKeys dump = new DumpKeys();
+            String actionsAt;
             try (Locker.Lock lock = _locker.lock())
             {
+                actionsAt = StdErrLog.timestamp(System.currentTimeMillis(),TimeUnit.MILLISECONDS);
```

Comment:
We can format a date using `java.time` from JDK 8.

Reply:
That will result in a different timestamp and also will not use the date formatting cache used by stderrlog.

Reply:
What different timestamp ? And why we should care at all about what `StdErrLog` does ? We change `StdErrLog` we break here ? I don't understand why we cannot use the JDK to format a date/time.

Reply:
I don't want to use JDK to format the time here as it will be a different format to what we are using in our logging and it will not use our dateCache, which is designed to help with formatting many similar times per second.

I'll move the method to be on Log, which will always be a public class.

Reply:
This is a JMX method. It does not get logged. It is reported in the JMX console where people copy&paste it into an editor. It does not have to have anything that is related to out logging implementation. And besides, most of the deployments will use another logging system that is not ours, so the format of the date will be different anyway.
I'm against this, and moving it to Log is even worse because this is a formatting concern now exposed into a public class.